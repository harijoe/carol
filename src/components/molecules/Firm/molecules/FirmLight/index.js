import React from 'react'
import PropTypes from 'prop-types'
import styled, { css } from 'styled-components'
import { FormattedMessage } from 'react-intl'
import { theme, breakpoint } from 'utils/style'

import { Heading, Paragraph, Icon, ProfileImage } from 'components'

const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
`
const HeaderWrapper = styled.header`
  ${({ imageUrl }) => css`
  position: relative;
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
  padding: ${theme('spaces.m')};
  height: 12.5rem;
  background: url(${imageUrl}) no-repeat top center;
  background-size: cover;

  &::before {
    position: absolute;
    top: 0;
    left: 0;
    z-index: 1;
    height: 100%;
    width: 100%;
    background-color: rgba(38, 38, 38, 0.5);
    content: '';
  }

  ${breakpoint('l')`
    padding: ${theme('spaces.l')};
  `}
`};
`

const TopHeaderWrapper = styled.div`
  position: relative;
  z-index: 1;
  display: flex;
  flex-wrap: nowrap;
  align-items: flex-end;
  justify-content: space-between;
`

const NoteFirm = styled.div`
  margin: 0;
  text-align: right;
  line-height: 1;
  color: ${theme('colors.white')};

  p {
    margin: 0;
  }

  strong {
    font-weight: bold;
    font-size: ${theme('fonts.size.xxl')};
  }

  span {
    font-size: ${theme('fonts.size.s')};
  }
`

const StyledProfileImage = styled(ProfileImage)`
  height: 8rem;
  width: 8rem;

  ${breakpoint('l')`
    margin-bottom: -3.6rem;
  `}
`

const StyledHeading = styled(Heading)`
  margin-bottom: ${theme('spaces.l')};
  font-size: ${theme('fonts.size.xl')};
  text-transform: lowercase;

  &::first-letter {
    text-transform: uppercase;
  }
`

const FooterWrapper = styled.footer`
  position: relative;
  display: flex;
  flex-direction: column;
  padding: ${theme('spaces.m')};

  ${breakpoint('l')`
    padding: ${theme('spaces.l')};
  `};
`

const StyledParagraph = styled(Paragraph)`
  margin-top: 0;
  color: ${theme('colors.grayscale.dark')};
`

const StyledIcon = styled(Icon)`
  margin-right: ${theme('spaces.s')};
  vertical-align: sub;

  svg {
    fill: ${theme('colors.grayscale.dark')};
  }
`

/**
 * Transforms an int or decimal into format notation
 *
 * ex: 4 => 4.0, 3.555 => 3.56, 4.6 => 4.6
 *
 * @param value
 */
const formatNotation = value => value.toFixed(`${value}`.split('.')[1] ? Math.min(`${value}`.split('.')[1].length, 2) : 1)

const FirmLight = ({ imageUrl, logoUrl, globalRating, name, globalRatingCount, postalCode }) =>
  <Wrapper>
    <HeaderWrapper {...{ imageUrl }}>
      <TopHeaderWrapper>
        <StyledProfileImage imageLink={logoUrl} />
        <NoteFirm>
          {globalRatingCount
            ? <p>
                <strong>{formatNotation(globalRating)}</strong>/5{' '}
              </p>
            : ''}
          {globalRatingCount
            ? <span>
                {globalRatingCount} <FormattedMessage id="firm.details.reviews" />
              </span>
            : ''}
        </NoteFirm>
      </TopHeaderWrapper>
    </HeaderWrapper>
    <FooterWrapper>
      <StyledHeading level={3}>
        {name}
      </StyledHeading>
      <StyledParagraph>
        <StyledIcon icon="location-pin" />
        {postalCode}
      </StyledParagraph>
    </FooterWrapper>
  </Wrapper>

FirmLight.propTypes = {
  name: PropTypes.string.isRequired,
  imageUrl: PropTypes.string,
  logoUrl: PropTypes.string.isRequired,
  postalCode: PropTypes.string.isRequired,
  globalRating: PropTypes.number,
  globalRatingCount: PropTypes.number,
}

FirmLight.defaultProps = {
  globalRating: null,
  globalRatingCount: null,
  imageUrl:
    'https://res.cloudinary.com/quotatis/image/upload/v1495120789/GB/ChatbotImages/Q1/experts-and-advices-before-work.jpg' /* @TODO: To remove when dora will provide it */,
}

export default FirmLight
