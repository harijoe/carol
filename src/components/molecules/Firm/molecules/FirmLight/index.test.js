import React from 'react'
import { shallow } from 'enzyme'

import FirmLight from './'

const props = {
  name: 'test',
  logoUrl: '',
  postalCode: '96352',
  globalRating: 4.5,
  globalRatingCount: 88,
}

it('renders the FirmLight component', () => {
  const wrapper = shallow(<FirmLight {...props} />)

  expect(wrapper).toMatchSnapshot()
})
