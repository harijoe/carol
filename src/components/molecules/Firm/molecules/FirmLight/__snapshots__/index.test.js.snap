// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the FirmLight component 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <styled.div>
    <styled.header
        imageUrl="https://res.cloudinary.com/quotatis/image/upload/v1495120789/GB/ChatbotImages/Q1/experts-and-advices-before-work.jpg"
    >
        <styled.div>
            <Styled(Mock)
                imageLink=""
            />
            <styled.div>
                <p>
                    <strong>
                        4.5
                    </strong>
                    /5
                     
                </p>
                <span>
                    88
                     
                    <FormattedMessage
                        id="firm.details.reviews"
                        values={Object {}}
                    />
                </span>
            </styled.div>
        </styled.div>
    </styled.header>
    <styled.footer>
        <Styled(Mock)
            level={3}
        >
            test
        </Styled(Mock)>
        <Styled(Mock)>
            <Styled(Mock)
                icon="location-pin"
            />
            96352
        </Styled(Mock)>
    </styled.footer>
</styled.div>,
  "nodes": Array [
    <styled.div>
      <styled.header
            imageUrl="https://res.cloudinary.com/quotatis/image/upload/v1495120789/GB/ChatbotImages/Q1/experts-and-advices-before-work.jpg"
      >
            <styled.div>
                  <Styled(Mock)
                        imageLink=""
                  />
                  <styled.div>
                        <p>
                              <strong>
                                    4.5
                              </strong>
                              /5
                               
                        </p>
                        <span>
                              88
                               
                              <FormattedMessage
                                    id="firm.details.reviews"
                                    values={Object {}}
                              />
                        </span>
                  </styled.div>
            </styled.div>
      </styled.header>
      <styled.footer>
            <Styled(Mock)
                  level={3}
            >
                  test
            </Styled(Mock)>
            <Styled(Mock)>
                  <Styled(Mock)
                        icon="location-pin"
                  />
                  96352
            </Styled(Mock)>
      </styled.footer>
</styled.div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <FirmLight
        globalRating={4.5}
        globalRatingCount={88}
        imageUrl="https://res.cloudinary.com/quotatis/image/upload/v1495120789/GB/ChatbotImages/Q1/experts-and-advices-before-work.jpg"
        logoUrl=""
        name="test"
        postalCode="96352"
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "globalRating": 4.5,
          "globalRatingCount": 88,
          "imageUrl": "https://res.cloudinary.com/quotatis/image/upload/v1495120789/GB/ChatbotImages/Q1/experts-and-advices-before-work.jpg",
          "logoUrl": "",
          "name": "test",
          "postalCode": "96352",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <styled.div>
          <styled.header
                    imageUrl="https://res.cloudinary.com/quotatis/image/upload/v1495120789/GB/ChatbotImages/Q1/experts-and-advices-before-work.jpg"
          >
                    <styled.div>
                              <Styled(Mock)
                                        imageLink=""
                              />
                              <styled.div>
                                        <p>
                                                  <strong>
                                                            4.5
                                                  </strong>
                                                  /5
                                                   
                                        </p>
                                        <span>
                                                  88
                                                   
                                                  <FormattedMessage
                                                            id="firm.details.reviews"
                                                            values={Object {}}
                                                  />
                                        </span>
                              </styled.div>
                    </styled.div>
          </styled.header>
          <styled.footer>
                    <Styled(Mock)
                              level={3}
                    >
                              test
                    </Styled(Mock)>
                    <Styled(Mock)>
                              <Styled(Mock)
                                        icon="location-pin"
                              />
                              96352
                    </Styled(Mock)>
          </styled.footer>
</styled.div>,
        "_debugID": 2,
        "_renderedOutput": <styled.div>
          <styled.header
                    imageUrl="https://res.cloudinary.com/quotatis/image/upload/v1495120789/GB/ChatbotImages/Q1/experts-and-advices-before-work.jpg"
          >
                    <styled.div>
                              <Styled(Mock)
                                        imageLink=""
                              />
                              <styled.div>
                                        <p>
                                                  <strong>
                                                            4.5
                                                  </strong>
                                                  /5
                                                   
                                        </p>
                                        <span>
                                                  88
                                                   
                                                  <FormattedMessage
                                                            id="firm.details.reviews"
                                                            values={Object {}}
                                                  />
                                        </span>
                              </styled.div>
                    </styled.div>
          </styled.header>
          <styled.footer>
                    <Styled(Mock)
                              level={3}
                    >
                              test
                    </Styled(Mock)>
                    <Styled(Mock)>
                              <Styled(Mock)
                                        icon="location-pin"
                              />
                              96352
                    </Styled(Mock)>
          </styled.footer>
</styled.div>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <FirmLight
    globalRating={4.5}
    globalRatingCount={88}
    imageUrl="https://res.cloudinary.com/quotatis/image/upload/v1495120789/GB/ChatbotImages/Q1/experts-and-advices-before-work.jpg"
    logoUrl=""
    name="test"
    postalCode="96352"
/>,
}
`;
